www/include/configuration/configCentreonBroker/formCentreonBroker.php

```diff
@@ -58,6 +58,40 @@
 
 $cbObj = new CentreonConfigCentreonBroker($pearDB);
 
+/**
+ * @param array<string,mixed> $data
+ * @return array<string,mixed>
+ */
+function htmlEncodeBrokerInformation(array $data): array
+{
+    $data['name'] = htmlentities($data['name']);
+    $data['filename'] = htmlentities($data['filename']);
+    $data['cache_directory'] = htmlentities($data['cache_directory']);
+    $data['log_directory'] = htmlentities($data['log_directory']);
+    $data['log_filename'] = htmlentities($data['log_filename']);
+    $data['bbdo_version'] = htmlentities($data['bbdo_version']);
+    $data['command_file'] = htmlentities($data['command_file']);
+
+    return $data;
+}
+
+/**
+ * @param array<string,mixed> $data
+ * @return array<string,mixed>
+ */
+function htmlDecodeBrokerInformation(array $data): array
+{
+    $data['name'] = html_entity_decode($data['name']);
+    $data['filename'] = html_entity_decode($data['filename']);
+    $data['cache_directory'] = html_entity_decode($data['cache_directory']);
+    $data['log_directory'] = html_entity_decode($data['log_directory']);
+    $data['log_filename'] = html_entity_decode($data['log_filename']);
+    $data['bbdo_version'] = html_entity_decode($data['bbdo_version']);
+    $data['command_file'] = html_entity_decode($data['command_file']);
+
+    return $data;
+}
+
```

Comment:
```suggestion
/**
 * @param array<string,mixed> $data
 * @return array<string,mixed>
 */
function htmlEncodeBrokerInformation(array $data): array
{
    return array_map('encodeToHTMLEntities', $data);
}

/**
 * @param array<string,mixed> $data
 * @return array<string,mixed>
 */
function htmlDecodeBrokerInformation(array $data): array
{
    return array_map('decodeHTMLEntity', $data);
}

function encodeToHTMLEntities(string $text): string
{
    return htmlentities($text);
}

function decodeHTMLEntity(string $htmlEntity): string
{
    return html_entity_decode($htmlEntity);
}

```

is more compact and elegant I think. However missing array keys would not be initialized.